austation/code/game/materialization.dm

```diff
@@ -0,0 +1,297 @@
+	for(var/mob/living/L as() in players)
+		if(QDELETED(L) || L.stat == DEAD)
+			remove_player(L)
+			continue
+	time_left -= delta_time SECONDS
+	var/rounded = round(time_left, 1)
+	if(rounded == 15 || rounded == 10 || rounded == 5)
+		say("[rounded]...")
+
+	if(time_left <= 0)
+		visible_message("<span class='warning'>[current_player] has ran out of time!</span>")
+		playsound(src, 'sound/effects/clock_tick.ogg', 120, FALSE, 2)
+		remove_player(current_player)
+
+/obj/structure/table/mat_shiritori/examine()
+	. = ..()
+	if(length(current_letter)) // because "" is not falsy
+		. += "<span class='info'>The next word must start with <b>[uppertext(current_letter)]</b></span>"
+	else if(!active)
+		. += "<span class='info'>It doesn't look like anyone is playing it right now.</span>"
+	. += "\n<span class='info'>[length(knockouts)] players have been eliminated. Turn [turns].</span>"
+
+// Dynamic list initialization, saves memory
+/obj/structure/table/mat_shiritori/proc/setup_blacklist()
+	if(!blacklist)
+		// This is not typesof, each path blacklists that atom only, good for root datums that have no functionality
+		blacklist = typecacheof(list(
+			/obj/structure/table/mat_shiritori,
+			/obj/item/shiritori_ball,
+			/obj/effect/decal/cleanable,
+			/obj/item/radio/headset,
+			/obj/item/clothing/head/helmet/space,
+			/obj/item/book/manual,
+			/obj/item/reagent_containers/food/drinks,
+			/obj/item/reagent_containers/food,
+			/obj/item/reagent_containers,
+			/obj/machinery/atmospherics,
+			/obj/machinery/portable_atmospherics,
+			/obj/item/mecha_parts/mecha_equipment/weapon/ballistic/missile_rack,
+			/obj/item/mecha_parts/mecha_equipment,
+			/obj/item/storage,
+			/obj/item/clothing,
+			/obj/item/stock_parts,
+			/obj/item/gun,
+			/obj/item/organ,
+			/obj/item,
+			/obj/machinery/power,
+			/obj/machinery,
+			/obj/effect,
+			/obj,
+			/mob/living/carbon,
+			/mob/living/simple_animal,
+			/mob/living,
+			/mob
+			), FALSE, TRUE)
+
+		// This one DOES include subtypes, this should contain items that can circumvent the game's rules, make it really unfun or just break the server
+		blacklist += typecacheof(list(
+			/obj/singularity,
+			/obj/item/projectile/hvp,
+			/obj/item/reagent_containers/food/snacks/store/bread/recycled,
+			/obj/machinery/portable_atmospherics,
+			/obj/item/uplink,
+			/obj/machinery/nuclearbomb
+		))
+
+// -------- Shiritori Ball, used to spawn the selected atom --------
+
+
+/obj/item/shiritori_ball
+	name = "materialization ball"
+	desc = "Materializes selected entities at it's location after a short delay."
+	icon = 'austation/icons/obj/items.dmi'
+	icon_state = "shiri_ball"
+	var/entity_path
+	var/countdown = 5
+	var/obj/structure/table/mat_shiritori/table
+	var/mob/living/carbon/human/owner
+
+/obj/item/shiritori_ball/proc/prime_spawn(obj/structure/table/mat_shiritori/table, mob/living/carbon/human/owner, entity_path)
+	src.entity_path = entity_path
+	src.table = table
+	src.owner = owner
+	spawnit()
```

Comment:
Why not make this New()? There's no need for a snowflake proc

Reply:
This is fine for datums but atom's pass all of their New() arguments into a few procs so it's generally a good idea to keep it clean where you can